kcidb/test_oo.py

```diff
@@ -0,0 +1,101 @@
+
+from kcidb.test_orm import parse
+
+
+class KCIDB00ClientTestCase(kcidb.unittest.TestCase):
+
+    def _pre_setup(self):
+        self.path = 'sqlite:test_oo.sqlite3'
+        self.source = db.Client(self.path)
+
+        self.initialize_and_load_data()
+        self.assertTrue(isinstance(self.source, kcidb.orm.Source))
+
+    def setUp(self):
+        self._pre_setup()
+        self.client = oo.Client(self.source)
+
+    def tearDown(self):
+        _, file = self.path.split(":")
+        os.unlink(file)
+
+    def initialize_and_load_data(self):
+        if not self.source.is_initialized():
+            self.source.init()
+
+        self.source.load(
+            {
+                "version": {
+                    "major": 3,
+                    "minor": 0
+                },
+                "tests": [
+                    {
+                        "waived": False,
+                        "start_time": "2021-11-23T03:52:13.660000+00:00",
+                        "path": "baseline.login",
+                        "status": "PASS",
+                        "id": "kernelci:kernelci.org:619c656de1fb4af479f2efaa",
+                        "build_id": "kernelci:kernelci.org:619c64b1712847eccbf2efac",
+                        "origin": "kernelci"
+                    },
+                    {
+                        "waived": False,
+                        "start_time": "2021-11-23T03:52:13.666000+00:00",
+                        "path": "baseline.dmesg.crit",
+                        "status": "PASS",
+                        "id": "kernelci:kernelci.org:619c656de1fb4af479f2efac",
+                        "build_id": "kernelci:kernelci.org:619c64b1712847eccbf2efac",
+                        "origin": "kernelci"
+                    },
+                    {
+                        "waived": False,
+                        "start_time": "2021-11-23T03:52:13.668000+00:00",
+                        "path": "baseline.dmesg.alert",
+                        "status": "PASS",
+                        "id": "kernelci:kernelci.org:619c656de1fb4af479f2efad",
+                        "build_id": "kernelci:kernelci.org:619c64b1712847eccbf2efac",
+                        "origin": "kernelci"
+                    },
+                    {
+                        "waived": False,
+                        "start_time": "2021-11-23T03:52:13.671000+00:00",
+                        "path": "baseline.dmesg.emerg",
+                        "status": "PASS",
+                        "id": "kernelci:kernelci.org:619c656de1fb4af479f2efae",
+                        "build_id": "kernelci:kernelci.org:619c64b1712847eccbf2efac",
+                        "origin": "kernelci"
+                    }
+                ],
+            }
+        )
+
+    def test_simple_build(self):
+        self.assertEqual(
+            self.client.query(
+                set(kcidb.orm.Pattern.parse('>test%#', [{('kernelci:kernelci.org:619c656de1fb4af479f2efac',)}]))
```

Comment:
`kcidb.orm.Pattern.parse()` already returns a set, so no need to call `set()` here.